MB-12440: Rename HandlerContext to HandlerConfig

ahobson opened the pull request:
## [MB-12440](https://dp3.atlassian.net/browse/MB-12440) for this change

## Summary

In go a [context](https://pkg.go.dev/context) has a very specific meaning. In retrospect, the currently named `HandlerContext` is not a context, but is really a config that is initialized once at server startup.

Rename `HandlerContext` -> `HandlerConfig` (and also rename variables that store the new `HandlerConfig`)

This was done almost entirely with automation, but a few of the variable names were tweaked by hand.

## Setup to Run Your Code

```
make server_test
```

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

Because this is a rename, there seems little chance that this would cause a regression if the code compiles.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

sandy-wright commented:
I found some files that still reference HandlerContext:

<img width="685" alt="Screen Shot 2022-05-10 at 12 16 48 PM" src="https://user-images.githubusercontent.com/62574202/167685552-e29e642a-1cf8-4040-926e-dcc49f46e485.png">

Most of them are comments, but some are function names that could be confusing if we left them. Otherwise, looks like the change is sound and tests work locally!

ahobson commented:
> I found some files that still reference HandlerContext:
...
> Most of them are comments, but some are function names that could be confusing if we left them. Otherwise, looks like the change is sound and tests work locally!

Thank you! I believe I've fixed all of them and rebased onto the latest master
